superwall/src/main/java/com/superwall/sdk/paywall/vc/SuperwallPaywallActivity.kt

```diff
@@ -90,16 +97,72 @@ class SuperwallPaywallActivity : AppCompatActivity() {
 
     private var contentView: View? = null
     private var notificationPermissionCallback: NotificationPermissionCallback? = null
+    private val isBottomSheetView
+        get() = contentView is CoordinatorLayout && Build.VERSION.SDK_INT >= Build.VERSION_CODES.R
 
     override fun setContentView(view: View) {
         super.setContentView(view)
         contentView = view
     }
 
+    private val mainScope = CoroutineScope(Dispatchers.Main)
+
+    private fun paywallView(): PaywallView? {
+        val content = contentView ?: error("Content view is null")
+        if (content is PaywallView) {
+            return content
+        } else if (content is CoordinatorLayout) {
+            return (content.getChildAt(0) as FrameLayout).getChildAt(0) as PaywallView
+        } else {
+            return null
+        }
+    }
+
+    private fun setupBottomSheetLayout(paywallView: PaywallView) {
+        val activityView =
+            layoutInflater.inflate(com.superwall.sdk.R.layout.activity_bottom_sheet, null)
+        setContentView(activityView)
+        initBottomSheetBehavior()
+        val container =
+            activityView.findViewById<FrameLayout>(com.superwall.sdk.R.id.container)
+        container.setOnClickListener { finish() }
+        container.addView(paywallView)
+        container.requestLayout()
+    }
+
+    private fun initBottomSheetBehavior() {
+        var bottomSheetBehavior = BottomSheetBehavior.from((contentView as ViewGroup).getChildAt(0))
+        bottomSheetBehavior.halfExpandedRatio = 0.62f
```

Comment:
Need to change to 0.7

Reply:
Updated, also removed some logs and updated how the paywall view is acquired 👍

Reply:
awesome!